### PR TITLE
Rename dataset to avoid ambiguous index pattern generation

### PR DESCRIPTION
**Pull request by jalvz**
## Motivation/summary

Solves the issue described in https://github.com/elastic/kibana/issues/88307#issuecomment-770913428

While we could maybe find a way out with priority ordering, that would introduce unnecessary complexity. 

Happy to replace "user" with something more appropriate if there are better ideas. 

## How to test these changes

Wait for package 0.5.0 release

### CODE REVIEW

File: apmpackage/apm/0.1.0/data_stream/app_metrics/manifest.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 title: APM application metrics
 type: metrics
-dataset: apm
+dataset: apm.user
```
Comment (axw) [inline apmpackage/apm/0.1.0/data_stream/app_metrics/manifest.yml:3]:
```suggestion
dataset: apm.app
```

The title of the stream is "app metrics" after all :)

**Review by axw — CHANGES_REQUESTED** [review]
Thanks, I'd forgotten about that one.

Please delete the checklist section in the description.

**Comment by simitt** [comment]
This PR fixes the apm issue; we still need to find a more generic answer to the question around how to avoid such conflicts right?

File: apmpackage/apm/0.1.0/data_stream/app_metrics/manifest.yml (hunk 1 above)
Comment (jalvz) [reply apmpackage/apm/0.1.0/data_stream/app_metrics/manifest.yml:3]: ha, I was 50/50 between "app" and "user"

**Review by axw — APPROVED** [review]

**Comment by jalvz** [comment]
> This PR fixes the apm issue; we still need to find a more generic answer to the question around how to avoid such conflicts right?

There is no much about it, if one dataset name is prefix of another, it is going to be ambiguous. Maybe it can be validated in the package registry.

**Comment by axw** [comment]
Confirmed with BC2 and epr-snapshot. All APM index templates have a prefix of `apm.` on the dataset:

image

Application metrics are being put into `metrics-apm.app.<service>-*` data streams:

image

**Pull request merged**